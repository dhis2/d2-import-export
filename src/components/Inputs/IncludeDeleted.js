import React from 'react'
import PropTypes from 'prop-types'
import i18n from '@dhis2/d2-i18n'
import { Switch } from '../Switch'

const NAME = 'includeDeleted'
const DATATEST = 'input-include-deleted'
const LABEL = i18n.t('Include deleted')

const IncludeDeleted = ({ value }) => (
    <Switch name={NAME} label={LABEL} value={value} dataTest={DATATEST} />
)

IncludeDeleted.propTypes = {
    value: PropTypes.bool.isRequired,
}

export { IncludeDeleted }
