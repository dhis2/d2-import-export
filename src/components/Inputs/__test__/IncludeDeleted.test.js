import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { IncludeDeleted } from '../IncludeDeleted'

jest.mock('react-final-form')

describe('Input component - IncludeDeleted', () => {
    it('should render correctly', () => {
        const file = shallow(<IncludeDeleted />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
