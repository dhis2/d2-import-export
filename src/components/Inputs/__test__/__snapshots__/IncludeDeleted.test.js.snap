// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - IncludeDeleted should render correctly 1`] = `
<Field>
  <Label>
    Include deleted
  </Label>
  <RadioGroup
    label="Include deleted"
    name="includeDeleted"
    options={
      Array [
        Object {
          "label": "Yes",
          "value": "true",
        },
        Object {
          "label": "No",
          "value": "false",
        },
      ]
    }
  />
</Field>
`;
